pkgs/applications/audio/ashuffle/default.nix

```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchgit, cmake, git, pkg-config, mpd_clientlib, meson, ninja }:
+
+stdenv.mkDerivation rec {
+  pname = "ashuffle";
+  version = "3.4.0";
+
+  src = fetchgit {
+    url = "https://github.com/joshkunz/ashuffle";
+    deepClone = true;
+    rev = "v${version}";
+    sha256 = "1k295fhhjq9n6ag605qzgxpfrr0h0xi9aggph3cy5s5i81pg7c92";
+  };
+
+  nativeBuildInputs = [ pkg-config meson ];
+  buildInputs = [ cmake git mpd_clientlib ninja ];
```

Comment:
ninja should likely be a `nativeBuildInput` as should `cmake`. As a side note it seems odd that this program uses both `cmake` and `meson`. Also `git` should probably be a `nativeBuildInput` if it is actually needed

Reply:
It doesn't build if `cmake` is in `nativeBuildInputs` - only `buildInputs`.  It appears I can move `ninja` without it affecting the build. And `git` is not needed, apparently.

I'm not sure about the difference between `buildInputs` and `nativeBuildInputs`, so sorry for the confusion.

Reply:
as a general rule `nativeBuildInputs` are needed at build time and `buildInputs` are needed when the program actually runs. This should only make a difference when cross compiling though. Is `cmake` actually needed? I expect when you move `cmake` to `nativeBuildInputs` it triggers its build hook (which probably breaks stuff)

Reply:
`nix-build -A ashuffle` fails when `cmake` is completely absent, or when it is in `nativeBuildInputs`. It only passes when `cmake` is present in `buildInputs`.

Reply:
This seems surprising, what error do you get when cmake isn't present

Reply:
Meson spits out some configuration, and then it blows up with:

```
|Executing subproject absl method cmake
|
|Did not find CMake 'cmake'
|Found CMake: NO

meson.build:37:4: ERROR: Unable to find CMake
```

So it's a submodule dependency that needs `cmake` to build.

Reply:
ah ok that makes sense, I still feel it should be a `nativeBuildInput`. Maybe there is someway to get cmake as a dependency without running the hook

Reply:
There's a flag on `mkDerivation` called `dontUseCmakeConfigure` which makes it work.

Thanks for the help.